packages/ui-components/src/components/TypingIndicator.tsx

```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import styled, { keyframes } from 'styled-components';
+
+// Keyframe animations
+const fadeIn = keyframes`
+  from { 
+    opacity: 0; 
+    transform: translateY(10px); 
+  }
+  to { 
+    opacity: 1; 
+    transform: translateY(0); 
+  }
+`;
+
+const pulse = keyframes`
+  0% {
+    opacity: 0.4;
+  }
+  50% {
+    opacity: 1;
+  }
+  100% {
+    opacity: 0.4;
+  }
+`;
+
+const TypingIndicatorContainer = styled.div`
+  display: flex;
+  align-items: center;
+  gap: 8px;
+  animation: ${fadeIn} 0.3s ease-in;
+  margin-left: 20px;
+`;
+
+const TypingBubble = styled.div`
+  background: ${props => props.theme.palette.common.white};
+  border-radius: 100px;
```

Comment:
A bit novel, but I think this shows intent more clearly:[^foo]

[^foo]: It’s on my list to add this as `--rounded` to `:root` at some point

```suggestion
  border-radius: calc(infinity * 1px);
```